main/document/add_link.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * This script allows to add cloud file links to the document structure
+ * @package chamilo.document
+ */
+/**
+ * Code
+ */
+ 
+// Including the global initialization file
+require_once '../inc/global.inc.php';
+ 
+// Including additional libraries
+require_once '../inc/lib/document.lib.php';
+require_once '../inc/lib/urlUtils.lib.php';
+
+$fileLinkEnabled = api_get_configuration_value('enable_add_file_link');
+
+if (!$fileLinkEnabled) {
+    api_not_allowed(true);
+}
+ 
+$course_info = api_get_course_info();
+ 
+if (empty($course_info)) {
+    api_not_allowed(true);
+}
+ 
+$document_data  = DocumentManager::get_document_data_by_id($_REQUEST['id'], api_get_course_id(), true);
+if (empty($document_data)) {
+    $document_id  = $parent_id =  0;
+    $path = '/';
+} else {
+    if ($document_data['filetype'] == 'folder') {
+        $document_id    = $document_data['id'];
+        $path           = $document_data['path']."/";
+        $parent_id      = DocumentManager::get_document_id(api_get_course_info(), dirname($path));
+    }
+}
+ 
+$is_allowed_to_edit = api_is_allowed_to_edit(null, true);
+ 
+if (api_get_group_id()) {
+    // If the group id is set, check if the user has the right to be here
+    // Needed for group related stuff
+    require_once api_get_path(LIBRARY_PATH).'groupmanager.lib.php';
+    // Get group info
+    $group_properties = GroupManager::get_group_properties(api_get_group_id());
+
+    if ($is_allowed_to_edit || GroupManager::is_user_in_group($_user['user_id'], api_get_group_id())) { // Only courseadmin or group members allowed
+        $to_group_id = api_get_group_id();
+        $req_gid = '&amp;gidReq='.api_get_group_id();
+        $interbreadcrumb[] = array('url' => '../group/group_space.php?gidReq='.api_get_group_id(), 'name' => get_lang('GroupSpace'));
+    } else {
+        api_not_allowed(true);
+    }
+} elseif ($is_allowed_to_edit || DocumentManager::is_my_shared_folder(api_get_user_id(), $path, api_get_session_id())) {
+    // Admin for "regular" upload, no group documents. And check if is my shared folder
+    $to_group_id = 0;
+    $req_gid = '';
+} else { // No course admin and no group member...
+    api_not_allowed(true);
+}
+ 
+// Group docs can only be uploaded in the group directory
+if ($to_group_id != 0 && $path == '/') {
+    $path = $group_properties['directory'] . "/";
+}
+ 
+// Breadcrumbs
+if ($is_certificate_mode) {
+    $interbreadcrumb[] = array('url' => '../gradebook/'.$_SESSION['gradebook_dest'], 'name' => get_lang('Gradebook'));
+} else {
+    $interbreadcrumb[] = array('url' => './document.php?id='.$document_id.$req_gid, 'name'=> get_lang('Documents'));
+}

```

Comment:
This condition about certificates is probably not necessary here. I mean... this only happens when dealing with certificates in the gradebook tool, and I don't think add_link.php should be accessible from there...
Maybe add a api_not_allowed() a few lines above if ($is_certificate_mode)